src/web/csp.rs

```diff
@@ -0,0 +1,189 @@
+use crate::config::Config;
+use iron::{AfterMiddleware, BeforeMiddleware, IronResult, Request, Response};
+
+pub(super) struct Csp {
+    nonce: String,
+    suppress: bool,
+}
+
+impl Csp {
+    fn new() -> Self {
+        let mut random = [0u8; 36];
+        getrandom::getrandom(&mut random).expect("failed to generate a nonce");
+        Self {
+            nonce: base64::encode(&random),
+            suppress: false,
+        }
+    }
+
+    pub(super) fn suppress(&mut self, suppress: bool) {
+        self.suppress = suppress;
+    }
```

Comment:
Mostly out of habit, and because this looks weird:

```rust
     req.extensions
        .get_mut::<Csp>()
        .expect("missing CSP")
        .suppress = true; 
```